From bb52bdd8a02934af545d6dcbf144506f5ad901c2 Mon Sep 17 00:00:00 2001
From: jmackay2 <jmackay2>
Date: Wed, 25 Jun 2025 19:27:02 -0400
Subject: [PATCH] thrust namespace for centroid transform

---
 gpu/features/src/centroid.cu | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gpu/features/src/centroid.cu b/gpu/features/src/centroid.cu
index 8ba0f80eefc..1dbd029cdf6 100644
--- a/gpu/features/src/centroid.cu
+++ b/gpu/features/src/centroid.cu
@@ -89,7 +89,7 @@ void pcl::device::compute3DCentroid(const DeviceArray<PointT>& cloud, float3& ce
     thrust::device_ptr<PointT> src_beg((PointT*)cloud.ptr());
     thrust::device_ptr<PointT> src_end = src_beg + cloud.size();
 
-    centroid = transform_reduce(src_beg, src_beg, PointT2float3<PointT>(), make_float3(0.f, 0.f, 0.f), PlusFloat3());
+    centroid = thrust::transform_reduce(src_beg, src_beg, PointT2float3<PointT>(), make_float3(0.f, 0.f, 0.f), PlusFloat3());
     centroid *= 1.f/cloud.size();
 }
 
@@ -105,7 +105,7 @@ void pcl::device::compute3DCentroid(const DeviceArray<PointT>& cloud, const Indi
         thrust::device_ptr<int> map_end = map_beg + indices.size();
 
 
-        centroid = transform_reduce(make_permutation_iterator(src_beg, map_beg),
+        centroid = thrust::transform_reduce(make_permutation_iterator(src_beg, map_beg),
             make_permutation_iterator(src_beg, map_end),
             PointT2float3<PointT>(), make_float3(0.f, 0.f, 0.f), PlusFloat3());
 
@@ -126,7 +126,7 @@ float3 pcl::device::getMaxDistance(const DeviceArray<PointT>& cloud, const float
     thrust::maximum<thrust::tuple<float, int>> op;
 
     thrust::tuple<float, int> res =
-        transform_reduce(
+        thrust::transform_reduce(
         make_zip_iterator(make_tuple( src_beg, cf )),
         make_zip_iterator(make_tuple( src_beg, ce )),
         TupleDistCvt(pivot), init, op);
@@ -153,7 +153,7 @@ float3 pcl::device::getMaxDistance(const DeviceArray<PointT>& cloud, const Indic
     thrust::tuple<float, int> init(0.f, 0);
     thrust::maximum<thrust::tuple<float, int>> op;
 
-    thrust::tuple<float, int> res = transform_reduce(
+    thrust::tuple<float, int> res = thrust::transform_reduce(
         make_zip_iterator(make_tuple( make_permutation_iterator(src_beg, map_beg), cf )),
         make_zip_iterator(make_tuple( make_permutation_iterator(src_beg, map_end), ce )),
         TupleDistCvt(pivot), init, op);
