From 64797959c950e7a9514b41abcbffca8bf7e0cd38 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Mon, 6 Jan 2025 19:41:43 +0100
Subject: [PATCH 1/4] fix CUDA

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 194a150..06c7edb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,14 +139,16 @@ endif()
 
 # CUDA support
 if (BUILD_CUDA_LIB)
-    find_package(CUDA)
-    if (CUDA_FOUND)
+    include(CheckLanguage)
+    check_language(CUDA)
+    if (CMAKE_CUDA_COMPILER)
+        enable_language(CUDA)
         message(STATUS "CUDA found (include: ${CUDA_INCLUDE_DIRS}, lib: ${CUDA_LIBRARIES})")
         include_directories(${CUDA_INCLUDE_DIRS})
-    else(CUDA_FOUND)
+    else()
         message(STATUS "CUDA not found, CUDA library will not be built")
         set(BUILD_CUDA_LIB OFF)
-    endif(CUDA_FOUND)
+    endif()
 endif(BUILD_CUDA_LIB)
 
 find_package(PkgConfig REQUIRED)
diff --git a/cmake/flann_utils.cmake b/cmake/flann_utils.cmake
index 1320aad..23d0024 100644
--- a/cmake/flann_utils.cmake
+++ b/cmake/flann_utils.cmake
@@ -99,7 +99,7 @@ endmacro(flann_add_gtest)
 
 macro(flann_add_cuda_gtest exe src)
     # add build target
-    cuda_add_executable(${exe} EXCLUDE_FROM_ALL ${src})
+    add_executable(${exe} EXCLUDE_FROM_ALL ${src})
     target_link_libraries(${exe} ${googletest_LIBRARIES} ${ARGN})
     add_dependencies(${exe} googletest)
 
diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 3ee6775..99fcd9b 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -37,7 +37,7 @@ if (BUILD_CUDA_LIB)
     else()
 	    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};" )
     endif()
-    cuda_add_library(flann_cuda_s STATIC ${CU_SOURCES})
+    add_library(flann_cuda_s STATIC ${CU_SOURCES})
     set_property(TARGET flann_cuda_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
     if (CMAKE_BUILD_STATIC_LIBS)
         list(APPEND flann_install_targets flann_cuda_s)
@@ -45,8 +45,9 @@ if (BUILD_CUDA_LIB)
         set_target_properties(flann_cuda_s PROPERTIES EXCLUDE_FROM_ALL true)
     endif()
 
-    cuda_add_library(flann_cuda SHARED ${CU_SOURCES})
+    add_library(flann_cuda SHARED ${CU_SOURCES})
     list(APPEND flann_install_targets flann_cuda)
+    set_property(TARGET flann_cuda PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
     set_property(TARGET flann_cpp PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
     set_property(TARGET flann_cpp_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC FLANN_USE_CUDA)
 else()
diff --git a/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu b/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu
index 4870253..6d06b65 100644
--- a/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu
+++ b/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu
@@ -572,7 +572,7 @@ int KDTreeCuda3dIndex<Distance >::radiusSearchGpu(const Matrix<ElementType>& que
 
     int buffer_index=0;
     for( size_t i=0; i<queries.rows; i++ ) {
-        for( size_t j=0; j<counts_host[i]; j++ ) {
+        for( int j=0; j<counts_host[i]; j++ ) {
             dists[i][j]=dists_temp[buffer_index];
             indices[i][j]=indices_temp[buffer_index];
             ++buffer_index;
diff --git a/test/flann_cuda_test.cu b/test/flann_cuda_test.cu
index d0aca0b..5162f6c 100644
--- a/test/flann_cuda_test.cu
+++ b/test/flann_cuda_test.cu
@@ -278,13 +278,13 @@ TEST_F(Flann_3D_Random_Cloud, Test4NN)
 TEST_F(Flann_3D_Random_Cloud, Test4NNGpuBuffers)
 {
 	thrust::host_vector<float4> data_host(data.rows);
-	for( int i=0; i<data.rows; i++ )
+	for( std::size_t i=0; i<data.rows; i++ )
 	{
 		data_host[i]=make_float4(data[i][0],data[i][1],data[i][2],0);
 	}
 	thrust::device_vector<float4> data_device = data_host;
 	thrust::host_vector<float4> query_host(data.rows);
-	for( int i=0; i<data.rows; i++ )
+	for( std::size_t i=0; i<data.rows; i++ )
 	{
 		query_host[i]=make_float4(query[i][0],query[i][1],query[i][2],0);
 	}
@@ -342,9 +342,9 @@ TEST_F(Flann_3D_Random_Cloud, TestRadiusSearchVector)
 	printf("done (%g seconds)", stop_timer());
 
 	start_timer("verifying results...");
-	for( int i=0; i<query.rows; i++ )
+	for( std::size_t i=0; i<query.rows; i++ )
 	{
-		for( int j=0; j<data.rows; j++ )
+		for( std::size_t j=0; j<data.rows; j++ )
 		{
 			float dist = 0;
 			for( int k=0; k<3; k++ )
@@ -367,9 +367,9 @@ TEST_F(Flann_3D_Random_Cloud, TestRadiusSearchVector)
 	printf("done (%g seconds)", stop_timer());
 	
 	start_timer("verifying results...");
-	for( int i=0; i<query.rows; i++ )
+	for( std::size_t i=0; i<query.rows; i++ )
 	{
-		for( int j=0; j<data.rows; j++ )
+		for( std::size_t j=0; j<data.rows; j++ )
 		{
 			// for each pair of query and data points: either the distance between them
 			// is smaller than r AND the point is in the result set, or 
@@ -406,7 +406,7 @@ TEST_F(Flann_3D_Random_Cloud, TestRadiusSearchMatrix)
 	printf("done (%g seconds)", stop_timer());
 	
 	int max_neighbors=0;
-	for( int i=0; i<query.rows; i++ )
+	for( std::size_t i=0; i<query.rows; i++ )
 	{
 		max_neighbors = std::max(max_neighbors, counts[i][0]);
 	}
@@ -419,9 +419,9 @@ TEST_F(Flann_3D_Random_Cloud, TestRadiusSearchMatrix)
 	printf("done (%g seconds)", stop_timer());
 
 	start_timer("verifying results...");
-	for( int i=0; i<query.rows; i++ )
+	for( std::size_t i=0; i<query.rows; i++ )
 	{
-		for( int j=0; j<data.rows; j++ )
+		for( std::size_t j=0; j<data.rows; j++ )
 		{
 			// for each pair of query and data points: either the distance between them
 			// is smaller than r AND the point is in the result set, or 
@@ -459,9 +459,9 @@ TEST_F(Flann_3D, TestRadiusSearch)
 	printf("done (%g seconds)\n", stop_timer());
 	
 	start_timer("verifying results...");
-	for( int i=0; i<query.rows; i++ )
+	for( std::size_t i=0; i<query.rows; i++ )
 	{
-		for( int j=0; j<data.rows; j++ )
+		for( std::size_t j=0; j<data.rows; j++ )
 		{
 			float dist = 0;
 			for( int k=0; k<3; k++ )
@@ -484,12 +484,12 @@ TEST_F(Flann_3D, TestRadiusSearch)
 	printf("done (%g seconds)\n", stop_timer());
 	
 	start_timer("verifying results...");
-	for( int i=0; i<query.rows; i++ )
+	for( std::size_t i=0; i<query.rows; i++ )
 	{
-		for( int j=0; j<data.rows; j++ )
+		for( std::size_t j=0; j<data.rows; j++ )
 		{
 			float dist = 0;
-			for( int k=0; k<3; k++ )
+			for( std::size_t k=0; k<3; k++ )
 				dist += (query[i][k]-data[j][k])*(query[i][k]-data[j][k]);
 			if( dist < r*r )
 			{
-- 
2.47.1


From 47628c74a3580b6c6f922344844a445413cdaa7a Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Mon, 6 Jan 2025 19:41:50 +0100
Subject: [PATCH 2/4] fix cmake

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 06c7edb..92bb82f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.28)
 
 if(COMMAND cmake_policy)
     cmake_policy(SET CMP0003 NEW)
-- 
2.47.1


From b2878811bc17d943703fbe93ad318afa1f2bf539 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Mon, 6 Jan 2025 19:42:02 +0100
Subject: [PATCH 3/4] fix python

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92bb82f..d743b5f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,8 +70,8 @@ endif()
 
 # find python
 if (BUILD_PYTHON_BINDINGS)
-find_package(PythonInterp)
-if (NOT PYTHON_EXECUTABLE)
+find_package (Python COMPONENTS Interpreter)
+if (NOT Python_EXECUTABLE)
     set(BUILD_PYTHON_BINDINGS OFF)
 endif()
 endif()
diff --git a/cmake/flann_utils.cmake b/cmake/flann_utils.cmake
index 23d0024..53c9c7d 100644
--- a/cmake/flann_utils.cmake
+++ b/cmake/flann_utils.cmake
@@ -124,7 +124,7 @@ macro(flann_add_pyunit file)
     string(REPLACE "/" "_" _testname ${file})
     add_test(
         NAME pyunit_${_testname}
-        COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/bin/run_test.py ${_file_name}
+        COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/bin/run_test.py ${_file_name}
         WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test
     )
     # add dependency to 'test' target
diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index 5057221..e20cbcf 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -5,8 +5,8 @@ install( FILES ${CMAKE_CURRENT_BINARY_DIR}/setup.py DESTINATION share/flann/pyth
 
 
 # python instalation
-if (PYTHON_EXECUTABLE)
+if (Python_EXECUTABLE)
     install(CODE "execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install
+        COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install
         WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
 endif()
-- 
2.47.1


From 1584e98647150292817aa2c51cb8ee9e238536b9 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Mon, 6 Jan 2025 19:42:09 +0100
Subject: [PATCH 4/4] fix latex

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/cmake/UseLATEX.cmake b/cmake/UseLATEX.cmake
index 77bd709..0ea3f26 100644
--- a/cmake/UseLATEX.cmake
+++ b/cmake/UseLATEX.cmake
@@ -751,8 +751,8 @@ MACRO(ADD_LATEX_TARGETS)
     ADD_CUSTOM_TARGET(${html_target}
       ${CMAKE_COMMAND} -E chdir ${output_dir}
       ${LATEX2HTML_CONVERTER} ${LATEX2HTML_CONVERTER_FLAGS} ${LATEX_MAIN_INPUT}
+      DEPENDS ${LATEX_MAIN_INPUT} ${LATEX_INPUTS}
       )
-    ADD_DEPENDENCIES(${html_target} ${LATEX_MAIN_INPUT} ${LATEX_INPUTS})
   ENDIF (LATEX2HTML_CONVERTER)
 
   ADD_CUSTOM_TARGET(${auxclean_target}
-- 
2.47.1

