From fcbe5b43dc1b1e232c8572777598d252701e71d3 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Thu, 25 Sep 2025 17:23:02 +0200
Subject: [PATCH] cassert

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/src/absolute_pose/CentralAbsoluteAdapter.cpp b/src/absolute_pose/CentralAbsoluteAdapter.cpp
index 684fa7e..ddaf4bf 100644
--- a/src/absolute_pose/CentralAbsoluteAdapter.cpp
+++ b/src/absolute_pose/CentralAbsoluteAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/absolute_pose/CentralAbsoluteAdapter.hpp>
 
+#include <cassert>
+
 
 opengv::absolute_pose::CentralAbsoluteAdapter::CentralAbsoluteAdapter(
     const bearingVectors_t & bearingVectors,
diff --git a/src/absolute_pose/MACentralAbsolute.cpp b/src/absolute_pose/MACentralAbsolute.cpp
index 6edbabc..c9629c2 100644
--- a/src/absolute_pose/MACentralAbsolute.cpp
+++ b/src/absolute_pose/MACentralAbsolute.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/absolute_pose/MACentralAbsolute.hpp>
 
+#include <cassert>
+
 
 opengv::absolute_pose::MACentralAbsolute::MACentralAbsolute(
     const double * points,
diff --git a/src/absolute_pose/MANoncentralAbsolute.cpp b/src/absolute_pose/MANoncentralAbsolute.cpp
index d9b5b09..29d6406 100644
--- a/src/absolute_pose/MANoncentralAbsolute.cpp
+++ b/src/absolute_pose/MANoncentralAbsolute.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/absolute_pose/MANoncentralAbsolute.hpp>
 
+#include <cassert>
+
 opengv::absolute_pose::MANoncentralAbsolute::MANoncentralAbsolute(
     const double * points,
     const double * bearingVectors,
diff --git a/src/absolute_pose/NoncentralAbsoluteAdapter.cpp b/src/absolute_pose/NoncentralAbsoluteAdapter.cpp
index 30176aa..d983ea6 100644
--- a/src/absolute_pose/NoncentralAbsoluteAdapter.cpp
+++ b/src/absolute_pose/NoncentralAbsoluteAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/absolute_pose/NoncentralAbsoluteAdapter.hpp>
 
+#include <cassert>
+
 opengv::absolute_pose::NoncentralAbsoluteAdapter::NoncentralAbsoluteAdapter(
     const bearingVectors_t & bearingVectors,
     const camCorrespondences_t & camCorrespondences,
diff --git a/src/absolute_pose/NoncentralAbsoluteMultiAdapter.cpp b/src/absolute_pose/NoncentralAbsoluteMultiAdapter.cpp
index 88c237a..50128ce 100644
--- a/src/absolute_pose/NoncentralAbsoluteMultiAdapter.cpp
+++ b/src/absolute_pose/NoncentralAbsoluteMultiAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/absolute_pose/NoncentralAbsoluteMultiAdapter.hpp>
 
+#include <cassert>
+
 opengv::absolute_pose::NoncentralAbsoluteMultiAdapter::NoncentralAbsoluteMultiAdapter(
     std::vector<std::shared_ptr<bearingVectors_t> > bearingVectors,
     std::vector<std::shared_ptr<points_t> > points,
diff --git a/src/absolute_pose/methods.cpp b/src/absolute_pose/methods.cpp
index 84c3ef4..80f2b7f 100644
--- a/src/absolute_pose/methods.cpp
+++ b/src/absolute_pose/methods.cpp
@@ -43,6 +43,8 @@
 #include <opengv/math/roots.hpp>
 
 #include <iostream>
+#include <cassert>
+
 
 opengv::translation_t
 opengv::absolute_pose::p2p(
diff --git a/src/absolute_pose/modules/main.cpp b/src/absolute_pose/modules/main.cpp
index a179aaf..1addbf3 100644
--- a/src/absolute_pose/modules/main.cpp
+++ b/src/absolute_pose/modules/main.cpp
@@ -46,6 +46,8 @@
 #include <opengv/math/arun.hpp>
 #include <opengv/math/cayley.hpp>
 
+#include <cassert>
+
 void
 opengv::absolute_pose::modules::p3p_kneip_main(
     const bearingVectors_t & f,
diff --git a/src/math/arun.cpp b/src/math/arun.cpp
index a0d6296..73b11ce 100644
--- a/src/math/arun.cpp
+++ b/src/math/arun.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/math/arun.hpp>
 
+#include <cassert>
+
 opengv::rotation_t
 opengv::math::arun( const Eigen::MatrixXd & Hcross )
 {
diff --git a/src/point_cloud/MAPointCloud.cpp b/src/point_cloud/MAPointCloud.cpp
index 81fd5dd..9f03933 100644
--- a/src/point_cloud/MAPointCloud.cpp
+++ b/src/point_cloud/MAPointCloud.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/point_cloud/MAPointCloud.hpp>
 
+#include <cassert>
+
 opengv::point_cloud::MAPointCloud::MAPointCloud(
     const double * points1,
     const double * points2,
diff --git a/src/point_cloud/PointCloudAdapter.cpp b/src/point_cloud/PointCloudAdapter.cpp
index f9faaeb..e71762c 100644
--- a/src/point_cloud/PointCloudAdapter.cpp
+++ b/src/point_cloud/PointCloudAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/point_cloud/PointCloudAdapter.hpp>
 
+#include <cassert>
+
 opengv::point_cloud::PointCloudAdapter::PointCloudAdapter(
     const points_t & points1,
     const points_t & points2 ) :
diff --git a/src/point_cloud/methods.cpp b/src/point_cloud/methods.cpp
index ed69488..023e520 100644
--- a/src/point_cloud/methods.cpp
+++ b/src/point_cloud/methods.cpp
@@ -39,6 +39,8 @@
 #include <opengv/math/arun.hpp>
 #include <opengv/math/cayley.hpp>
 
+#include <cassert>
+
 namespace opengv
 {
 namespace point_cloud
diff --git a/src/relative_pose/CentralRelativeAdapter.cpp b/src/relative_pose/CentralRelativeAdapter.cpp
index 38e9a62..682420a 100644
--- a/src/relative_pose/CentralRelativeAdapter.cpp
+++ b/src/relative_pose/CentralRelativeAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/CentralRelativeAdapter.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::CentralRelativeAdapter::CentralRelativeAdapter(
     const bearingVectors_t & bearingVectors1,
     const bearingVectors_t & bearingVectors2 ) :
diff --git a/src/relative_pose/CentralRelativeMultiAdapter.cpp b/src/relative_pose/CentralRelativeMultiAdapter.cpp
index 2ab7476..10c309d 100644
--- a/src/relative_pose/CentralRelativeMultiAdapter.cpp
+++ b/src/relative_pose/CentralRelativeMultiAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/CentralRelativeMultiAdapter.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::CentralRelativeMultiAdapter::CentralRelativeMultiAdapter(
     std::vector<std::shared_ptr<bearingVectors_t> > bearingVectors1,
     std::vector<std::shared_ptr<bearingVectors_t> > bearingVectors2 ) :
diff --git a/src/relative_pose/CentralRelativeWeightingAdapter.cpp b/src/relative_pose/CentralRelativeWeightingAdapter.cpp
index a6ab478..7baf205 100644
--- a/src/relative_pose/CentralRelativeWeightingAdapter.cpp
+++ b/src/relative_pose/CentralRelativeWeightingAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/CentralRelativeWeightingAdapter.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::CentralRelativeWeightingAdapter::CentralRelativeWeightingAdapter(
     const bearingVectors_t & bearingVectors1,
     const bearingVectors_t & bearingVectors2,
diff --git a/src/relative_pose/MACentralRelative.cpp b/src/relative_pose/MACentralRelative.cpp
index ec2959f..e1a2609 100644
--- a/src/relative_pose/MACentralRelative.cpp
+++ b/src/relative_pose/MACentralRelative.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/MACentralRelative.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::MACentralRelative::MACentralRelative(
     const double * bearingVectors1,
     const double * bearingVectors2,
diff --git a/src/relative_pose/MANoncentralRelative.cpp b/src/relative_pose/MANoncentralRelative.cpp
index cea9c14..e7679ae 100644
--- a/src/relative_pose/MANoncentralRelative.cpp
+++ b/src/relative_pose/MANoncentralRelative.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/MANoncentralRelative.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::MANoncentralRelative::MANoncentralRelative(
       const double * bearingVectors1,
       const double * bearingVectors2,
diff --git a/src/relative_pose/MANoncentralRelativeMulti.cpp b/src/relative_pose/MANoncentralRelativeMulti.cpp
index 49f8ecf..9386e8c 100644
--- a/src/relative_pose/MANoncentralRelativeMulti.cpp
+++ b/src/relative_pose/MANoncentralRelativeMulti.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/MANoncentralRelativeMulti.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::MANoncentralRelativeMulti::MANoncentralRelativeMulti(
     const std::vector<double*> & bearingVectors1,
     const std::vector<double*> & bearingVectors2,
diff --git a/src/relative_pose/NoncentralRelativeAdapter.cpp b/src/relative_pose/NoncentralRelativeAdapter.cpp
index 552f180..b4a9a14 100644
--- a/src/relative_pose/NoncentralRelativeAdapter.cpp
+++ b/src/relative_pose/NoncentralRelativeAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/NoncentralRelativeAdapter.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::NoncentralRelativeAdapter::NoncentralRelativeAdapter(
     const bearingVectors_t & bearingVectors1,
     const bearingVectors_t & bearingVectors2,
diff --git a/src/relative_pose/NoncentralRelativeMultiAdapter.cpp b/src/relative_pose/NoncentralRelativeMultiAdapter.cpp
index f41edbe..718e482 100644
--- a/src/relative_pose/NoncentralRelativeMultiAdapter.cpp
+++ b/src/relative_pose/NoncentralRelativeMultiAdapter.cpp
@@ -31,6 +31,8 @@
 
 #include <opengv/relative_pose/NoncentralRelativeMultiAdapter.hpp>
 
+#include <cassert>
+
 opengv::relative_pose::NoncentralRelativeMultiAdapter::NoncentralRelativeMultiAdapter(
     std::vector<std::shared_ptr<bearingVectors_t> > bearingVectors1,
     std::vector<std::shared_ptr<bearingVectors_t> > bearingVectors2,
diff --git a/src/relative_pose/methods.cpp b/src/relative_pose/methods.cpp
index 939ad6e..fbc704a 100644
--- a/src/relative_pose/methods.cpp
+++ b/src/relative_pose/methods.cpp
@@ -42,6 +42,8 @@
 #include <opengv/triangulation/methods.hpp>
 
 #include <iostream>
+#include <cassert>
+
 
 opengv::translation_t
 opengv::relative_pose::twopt(
diff --git a/src/relative_pose/modules/fivept_nister/modules.cpp b/src/relative_pose/modules/fivept_nister/modules.cpp
index b4f8d7d..b9e902c 100644
--- a/src/relative_pose/modules/fivept_nister/modules.cpp
+++ b/src/relative_pose/modules/fivept_nister/modules.cpp
@@ -35,6 +35,9 @@
 
 #include <opengv/OptimizationFunctor.hpp>
 
+#include <cassert>
+
+
 void
 opengv::relative_pose::modules::fivept_nister::composeA(
     const Eigen::Matrix<double,9,4> & EE,
-- 
2.51.0

