diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4aa5c3f8..9ccc375f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,8 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.1 FATAL_ERROR)
 
-INCLUDE(cmake/Utils.cmake)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+INCLUDE(Utils)
 
 INCLUDE(CheckIncludeFiles)
 INCLUDE(CheckIncludeFileCXX)
@@ -279,7 +280,7 @@ IF(VMIME_BUILD_TESTS)
 
 	ENABLE_TESTING()
 
-	INCLUDE(cmake/FindCppUnit.cmake)
+	INCLUDE(FindCppUnit)
 	INCLUDE_DIRECTORIES(${CPPUNIT_INCLUDE_DIR})
 
 	FILE(
@@ -626,7 +627,7 @@ OPTION(
 ##############################################################################
 # SASL support
 
-INCLUDE(cmake/FindGSasl.cmake)
+INCLUDE(FindGSasl)
 
 OPTION(
 	VMIME_HAVE_SASL_SUPPORT
@@ -743,12 +744,10 @@ ENDIF(VMIME_HAVE_TLS_SUPPORT)
 ##############################################################################
 # Charset conversion library
 
-INCLUDE(cmake/FindIconv.cmake)
-INCLUDE(cmake/FindICU.cmake)
+FIND_PACKAGE(ICU COMPONENTS uc REQUIRED)
+FIND_PACKAGE(Iconv REQUIRED)
 
-FIND_PACKAGE(ICU QUIET)
-
-IF(ICU_LIBRARIES)
+IF(_ICU_LIB_RELEASE_uc)
 	SET(VMIME_CHARSETCONV_LIB_DETECTED "icu")
 ELSEIF(ICONV_FOUND)
 	SET(VMIME_CHARSETCONV_LIB_DETECTED "iconv")
@@ -762,6 +761,7 @@ SET(
 	CACHE
 	STRING
 	"Library to use for charset conversion"
+	FORCE
 )
 SET_PROPERTY(
 	CACHE
