From d7922c0d242409c4ff8267aa738d2eb8a38c6795 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx@gmail.com>
Date: Fri, 30 May 2025 17:35:05 +0200
Subject: [PATCH 08/19] 2.5.59.13 fstream

Signed-off-by: Paul Zander <negril.nx@gmail.com>

diff --git a/src/slic3r/GUI/DesktopIntegrationDialog.cpp b/src/slic3r/GUI/DesktopIntegrationDialog.cpp
index 557c993d0..a4e8246f4 100644
--- a/src/slic3r/GUI/DesktopIntegrationDialog.cpp
+++ b/src/slic3r/GUI/DesktopIntegrationDialog.cpp
@@ -4,6 +4,7 @@
 ///|/ PrusaSlicer is released under the terms of the AGPLv3 or higher
 ///|/
 #ifdef __linux__
+#include <fstream>
 #include "DesktopIntegrationDialog.hpp"
 #include "GUI_App.hpp"
 #include "GUI.hpp"
diff --git a/src/slic3r/GUI/GUI_App.cpp b/src/slic3r/GUI/GUI_App.cpp
index d2e8eb370..22eeaecd8 100644
--- a/src/slic3r/GUI/GUI_App.cpp
+++ b/src/slic3r/GUI/GUI_App.cpp
@@ -26,6 +26,7 @@
 #include <algorithm>
 #include <iterator>
 #include <exception>
+#include <fstream>
 #include <cstdlib>
 #include <regex>
 #include <string_view>
-- 
2.49.0

