From cfcf675e57f39d2ca24e23040f020bd46319009b Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx@gmail.com>
Date: Sun, 15 Mar 2026 13:33:37 +0100
Subject: cmake: don't force CMAKE_CUDA_ARCHITECTURES

Change-Id: I2b9f6f749871a59c127f3a7ed8a2d797f5f33610
Signed-off-by: Paul Zander <negril.nx@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19ee2a86..9d4c2dd6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -324,6 +324,7 @@ if (USE_CUDA)
         "find_dependency(CUDAToolkit ${CUDAToolkit_VERSION})")
       enable_language(CUDA)
       if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.18")
+      if ( NOT CMAKE_CUDA_ARCHITECTURES )
         set(CMAKE_CUDA_ARCHITECTURES "")
         if (CUDAToolkit_VERSION VERSION_LESS "13.0")
           # Support Maxwell GPUs.
@@ -349,6 +350,7 @@ if (USE_CUDA)
           # Support Hopper GPUs.
           list(APPEND CMAKE_CUDA_ARCHITECTURES "90")
         endif(CUDAToolkit_VERSION VERSION_GREATER_EQUAL "11.8")
+      endif()
         message("-- Setting CUDA Architecture to ${CMAKE_CUDA_ARCHITECTURES}")
       endif()
 
-- 
2.53.0

